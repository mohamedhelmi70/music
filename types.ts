export type AppParamList = {
    Tab: undefined;
    Home: undefined;
    Track: {
        trackId: number;
    }
    Artists: undefined;
    Artist: {
        artistId: number;
    }
    Search: undefined;
    Albums: undefined;
    Album:  {
        albumId: number;
    }
};

export interface Album {
    album: string,
    id_album: number,
    cover: string,
    api_album: string,
    api_tracks: string
}

export interface Track {
    id_track: number;
    track: string;
    bpm: number;
    api_track: string;
    api_lyrics: string;
    artist: string;
}

export interface Artist {
    id_artist: number,
    artist: string,
    cover: string,
}