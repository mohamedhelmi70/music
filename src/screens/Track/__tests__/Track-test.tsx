import {RouteProp} from '@react-navigation/core';
import React from 'react';
import renderer from 'react-test-renderer';
import {AppParamList} from '../../../../types';
import TrackScreen from '../Track';

// Describing a test suite
describe('<Track />', () => {
    const route: RouteProp<AppParamList, 'Track'> = {
        params: {trackId: 121, albumId: 15, artistId: 12, title: 'track title'},
        key: 'q1',
        name: 'Track',
    };
    test('Renders Track correctly', () => {
        const tree = renderer.create(<TrackScreen route={route} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
