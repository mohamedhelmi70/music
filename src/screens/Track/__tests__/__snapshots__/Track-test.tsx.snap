// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Track /> Renders Track correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F7F7F9",
      },
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F7F7F9",
          "flex": 1,
          "justifyContent": "center",
          "width": "100%",
        },
        undefined,
      ],
    ]
  }
>
  <ActivityIndicator
    color="#03873B"
    size="small"
  />
</View>
`;
